### ISSUE TITLE
Surfaces in JupyterLab interface

### ISSUE THREAD

**Issue by robertodr** [nglviewer/nglview#975]
I think not all is well (yet) for the JupyterLab extension. Identical code for rendering a surface from a cube file works differently in the notebook and lab interfaces. Here are links to the [notebook](https://mybinder.org/v2/gh/ENCCS/nglview-demo/859be2a342bdd54f0d274a1a1c0a92df4e35022e?filepath=nglview-demo.ipynb) and [lab](https://mybinder.org/v2/gh/ENCCS/nglview-demo/859be2a342bdd54f0d274a1a1c0a92df4e35022e?urlpath=lab%2Ftree%2Fnglview-demo.ipynb) on MyBinder. 

- Version report

```bash
python -c 'import nglview; print(nglview.__version__)'
3.0.1
python -c 'import ipywidgets; print(ipywidgets.__version__)'
7.6.3
```

**Comment by hainm** [comment]
Here is from JS console:

<img width="329" alt="Screen Shot 2021-04-19 at 11 31 23 AM" src="https://user-images.githubusercontent.com/4451957/115262743-d033e300-a102-11eb-8beb-fdc8f36ee105.png">

**Comment by hainm** [comment]
um. I think it's due to this error: https://github.com/nglviewer/nglview/issues/929

If you copy the `HOMO.cube` to `/tmp` to avoid "current" path, the surface will be shown.

**Comment by robertodr** [comment]
> um. I think it's due to this error: #929
> 
> If you copy the `HOMO.cube` to `/tmp` to avoid "current" path, the surface will be shown.

Yes, if I copy it to `/tmp` and do `c1 = v1.add_component("/tmp/HOMO.cube")` then it works.

**Comment by hainm** [comment]
> um. I think it's due to this error: #929
> 
> If you copy the `HOMO.cube` to `/tmp` to avoid "current" path, the surface will be shown.

Um, this does not help if trying it in my mac. Weird. 

if I modify this line to NOT use filename, I am able to view a structure file (e.g: `./abc.pdb`) but I got error for the binary file (e.g: `./abc.dxbin`)
https://github.com/nglviewer/nglview/blob/b1d50de25672a3849ec2d0feb770c73f14de7b27/nglview/utils/py_utils.py#L224

**Comment by jaimergp** [comment]
We also seeing the same with `io.BytesIO` objects. It works on `jupyter notebook` but fails on `jupyter lab` with the [same error](https://github.com/nglviewer/nglview/issues/975#issuecomment-822562567) you provided.

```python

import gzip
import requests
import nglview
import io

selected_pocket_volume_url = 'https://proteins.plus/results/dogsite/DULB3XTM1tDVFafpkvYcAXJM/3w32_P_0_0_gpsAll.ccp4.gz'

r = requests.get(selected_pocket_volume_url)
r.raise_for_status()
# Decompress response content and wrap it in a file-like object
memfile = io.BytesIO(gzip.decompress(r.content))

viewer = nglview.show_pdbid("3w32")
# Since we are passing bytes with no path or extension
# we need to tell nglview what kind of data this is: ccp4
viewer.add_component(memfile, ext="ccp4")
viewer.center()
viewer
```

**Comment by Armin-Ariamajd** [comment]
Hi @hainm 

I'm also having the same problem; map files show on Jupyter notebook
<img width="1080" alt="Screenshot 2023-02-24 at 12 07 55" src="https://user-images.githubusercontent.com/80158628/221164398-c625d81f-b700-494e-a0a2-597cb70b583b.png">
but not on JupyterLab:
![Screenshot 2023-02-24 at 12 09 48](https://user-images.githubusercontent.com/80158628/221164752-fdc9bc24-fe2d-4ea7-a2bd-a5658693e765.png)
Is there a way to fix this? Thanks!

**Comment by hainm** [comment]
> Hi @hainm
> 
> I'm also having the same problem; map files show on Jupyter notebook <img alt="Screenshot 2023-02-24 at 12 07 55" width="1080" src="https://user-images.githubusercontent.com/80158628/221164398-c625d81f-b700-494e-a0a2-597cb70b583b.png"> but not on JupyterLab: ![Screenshot 2023-02-24 at 12 09 48](https://user-images.githubusercontent.com/80158628/221164752-fdc9bc24-fe2d-4ea7-a2bd-a5658693e765.png) Is there a way to fix this? Thanks!

dear @Armin-Ariamajd: Unfortunately I don't know the reason why.

**Comment by hainm** [comment]
> Hi @hainm
> 
> I'm also having the same problem; map files show on Jupyter notebook <img alt="Screenshot 2023-02-24 at 12 07 55" width="1080" src="https://user-images.githubusercontent.com/80158628/221164398-c625d81f-b700-494e-a0a2-597cb70b583b.png"> but not on JupyterLab: ![Screenshot 2023-02-24 at 12 09 48](https://user-images.githubusercontent.com/80158628/221164752-fdc9bc24-fe2d-4ea7-a2bd-a5658693e765.png) Is there a way to fix this? Thanks!

@Armin-Ariamajd 
This should work with latest `nglview` (3.0.5)

<img width="991" alt="image" src="https://github.com/nglviewer/nglview/assets/4451957/652a2a3b-86c1-4eb2-ab89-0b6b2b046569">

PS: You might want to use `view.add_surface` too,

**Comment by hainm** [comment]
The issue is resolved in #1100 (available in `nglview 3.1.1`)
